include/wil/registry.h

```diff
@@ -17,16 +17,3139 @@
 
 #include <winreg.h>
 #include <new.h> // new(std::nothrow)
-#include "resource.h" // unique_hkey
+#include "registry_helpers.h"
+#include "resource.h"
+
+#ifndef __WIL_WINREG_
+#error This is required for ::wil::unique_hkey support
+#endif
+
+// wil registry does not require the use of the STL or C++ exceptions
+// wil registry natively supports std::vector and std::wstring when preferring those types
+// wil registry uses the __WIL_WINREG_STL define to enable support for wil::shared_* types (defined in resource.h)
+// include sddl.h if wanting to pass security descriptor strings when creating keys
 
 namespace wil
 {
-    //! The key name includes the absolute path of the key in the registry, always starting at a
-    //! base key, for example, HKEY_LOCAL_MACHINE.
-    size_t const max_registry_key_name_length = 255;
+    namespace reg
+    {
+        /**
+         * \brief Helper function to translate registry return values if the value was not found
+         * \param hr HRESULT to test from registry APIs
+         * \return boolean if the HRESULT indicates the registry value was not found
+         */
+        constexpr bool is_hresult_not_found(HRESULT hr) WI_NOEXCEPT
+        {
+            return hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
+        }
+
+        /**
+         * \brief Helper function to translate registry return values if the buffer was too small
+         * \param hr HRESULT to test from registry APIs
+         * \return boolean if the HRESULT indicates the buffer was too small for the value being read
+         */
+        constexpr bool is_hresult_buffer_too_small(HRESULT hr) WI_NOEXCEPT
+        {
+            return hr == HRESULT_FROM_WIN32(ERROR_MORE_DATA);
+        }
+
+#if defined(WIL_ENABLE_EXCEPTIONS)
+        /**
+         * \brief Opens a new HKEY to the specified path

```

Comment:
Nit: I recommend using full sentence casing in the comments. VS Code gets rid of newlines (I have an image but I can't paste it into this UI lol)

Reply:
It's going to be unpleasant to force every Doxygen comment to be a full sentence that's grammatically correct.
Do I just need to put a period at the end of each comment so that VS Code is happy?
(I don't know how other editors behave, but that's a non-tractable problem)

Reply:
Yeah---I don't think we should contort the language or make it less concise. But we can't rely on newlines as separators.

Reply:
This be mostly resolved? Can you double-check the IntelliSense in VS?

Reply:
I should have fixed most of these.